client/src/view/container/observer/useObserver.tsx

```diff
@@ -0,0 +1,26 @@
+import { useSnackbar } from "notistack";
+import { Observer } from "rxjs";
+import { AjaxError } from "rxjs/ajax";
+
+export function useObserver<T>(callback: (params: T) => void): Observer<T> {
+  const { enqueueSnackbar } = useSnackbar();
+
+  return {
+    next: (params: T) => callback(params),
+    error: (err: Error) => handleError(err as AjaxError),
+    complete: () => {},
+  };
+
+  function handleError(err: AjaxError) {
+    if (err.status === 404)
+      enqueueSnackbar("Session error", {
+        preventDuplicate: true,
+        variant: "error",
+      });
+    else
+      enqueueSnackbar("Internal server error", {
```

Comment:
Šitą irgi galima iškelti į metodą, kur pasiduotum tik message

Reply:
Ir šiaip view neturėtų žinoti apie HTTP error codes. Exception'ai pas mus kaip ir modeliai konvertuojasi per layer'ius. Dabar jau neperdarinėk, nėra tikslo ir laiko, bet atsimink šitą